lib/opencensus/tags/formatters/binary.rb

```diff
@@ -0,0 +1,142 @@
+# frozen_string_literal: true
+
+module OpenCensus
+  module Tags
+    module Formatters
+      ##
+      # This formatter serializes and deserializes tags context according to
+      # the OpenCensus' BinaryEncoding specification. See
+      # [documentation](https://github.com/census-instrumentation/opencensus-specs/blob/master/encodings/BinaryEncoding.md).
+      #
+      # @example Serialize
+      #
+      # formatter = OpenCensus::Tags::Formatters::Binary.new
+      #
+      # tag_map = OpenCensus::Tags::TagMap.new({\"key1" => \"val1"})
+      # binary = formatter.serialize tag_map # "\x00\x00\x04key1\x04val1"
+      #
+      # @example Deserialize
+      #
+      # formatter = OpenCensus::Tags::Formatters::Binary.new
+      #
+      # binary = "\x00\x00\x04key1\x04val1"
+      # tag_map = formatter.deserialize binary
+      #
+      class Binary
+        # @private
+        class BinaryFormatterError < StandardError; end
+
+        # @private
+        #
+        # Seralization version
+        VERSION_ID = 0
+
+        # @private
+        #
+        # Tag field id
+        TAG_FIELD_ID = 0
+
+        # @private
+        #
+        # Serialized tag context limit
+        TAG_MAP_SERIALIZED_SIZE_LIMIT = 8192
+
+        # Serialize TagMap object
+        #
+        # @param [TagMap] tags_context
+        #
+        def serialize tags_context
+          binary = [int_to_varint(VERSION_ID)]
+
+          tags_context.each do |key, value|
+            binary << int_to_varint(TAG_FIELD_ID)
+            binary << int_to_varint(key.length)
+            binary << key.encode(Encoding::UTF_8)
+            binary << int_to_varint(value ? value.length : 0)
+            binary << value.to_s.encode(Encoding::UTF_8)
+          end
+
+          binary = binary.join
+          binary.length > TAG_MAP_SERIALIZED_SIZE_LIMIT ? nil : binary
+        end
+
+        # Deserialize binary data into a TagMap object.
+        #
+        # @param [String] binary
+        # @return [TagMap]
+        #
+        def deserialize binary
```

Comment:
Document that this can raise `BinaryFormatterError`.

Reply:
Added doc.